src/lighting/TrainLighting.cpp

```diff
@@ -0,0 +1,307 @@
+        generator_engine = p_generator_engine;
+        _dirty = true;
+    }
+
+    TrainEngine::EngineType TrainLighting::get_generator_engine() const {
+        return generator_engine;
+    }
+
+    void TrainLighting::set_max_accumulator_voltage(const double p_max_accumulator_voltage) {
+        max_accumulator_voltage = p_max_accumulator_voltage;
+        _dirty = true;
+    }
+
+    double TrainLighting::get_max_accumulator_voltage() const {
+        return max_accumulator_voltage;
+    }
+
+    void TrainLighting::set_alternative_light_source(const TrainController::TrainPowerSource p_light_source) {
+        alternative_light_source = p_light_source;
+        _dirty = true;
+    }
+
+    TrainController::TrainPowerSource TrainLighting::get_alternative_light_source() const {
+        return alternative_light_source;
+    }
+
+    void TrainLighting::set_alternative_max_voltage(const double p_max_accumulator_voltage) {
+        max_accumulator_voltage = p_max_accumulator_voltage;
+        _dirty = true;
+    }
+
+    double TrainLighting::get_alternative_max_voltage() const {
+        return alternative_max_voltage;
+    }
+
+    void TrainLighting::set_alternative_light_capacity(const double p_max_accumulator_voltage) {
+        max_accumulator_voltage = p_max_accumulator_voltage;
+        _dirty = true;
+    }
+
+    double TrainLighting::get_alternative_light_capacity() const {
+        return alternative_max_voltage;
+    }
+
+    void TrainLighting::set_accumulator_recharge_source(
+            const TrainController::TrainPowerSource p_accumulator_recharge_source) {
+        accumulator_recharge_source = p_accumulator_recharge_source;
+        _dirty = true;
+    }
+
+    TrainController::TrainPowerSource TrainLighting::get_accumulator_recharge_source() const {
+        return accumulator_recharge_source;
+    }
+
+    void TrainLighting::set_light_position_list(const TypedArray<LightListItem> &p_light_position_list) {
+        light_position_list.clear();
+        light_position_list.append_array(p_light_position_list);
+        _dirty = true;
+    }
+
+    TypedArray<LightListItem> TrainLighting::get_light_position_list() const {
+        return light_position_list;
+    }
+
+    void TrainLighting::set_wrap_light_selector(const bool p_wrap_light_selector) {
+        wrap_light_selector = p_wrap_light_selector;
+        _dirty = true;
+    }
+
+    bool TrainLighting::get_wrap_light_selector() const {
+        return wrap_light_selector;
+    }
+
+    void TrainLighting::set_default_selector_position(const int p_selector_position) {
+        default_selector_position = p_selector_position;
+        _dirty = true;
+    }
+
+    int TrainLighting::get_default_selector_position() const {
+        return default_selector_position;
+    }
+
+    void TrainLighting::set_selector_position(const int p_selector_position) {
+        selector_position = p_selector_position;
+        emit_signal("wrapper_position_changed", p_selector_position);
```

Comment:
1. DRY. Use `TrainLighting::WRAPPER_POSITION_CHANGED_SIGNAL` 
2. Consider changing the signal name, as mentioned above. "Wrapper" is not understandable in this context.